part_one_fibonacci.asm

```diff
@@ -0,0 +1,84 @@
+# COMP3410 Program Template
+# Author: Ben Murphy
+# Assignment: PA1
+# Date: Feb 10
+# part_one_fibonacci.asm
+# Input: Integer N from keyboard
+# Output: Print to terminal N Fibonacci numbers
+
+	.data #text for prompting for number of terms
+	
+prompt: .asciiz "How many Fibonacci numbers would you like to calculate?: "
+invalid: .asciiz "That number is invalid. Please make sure the number is greater than zero.\n"
+
+	.text
+beginning:	
+li $v0, 4			#we're about to print the prompt
+la $a0, prompt			#prime the prompt
+syscall			#print it
+li $v0, 5			#prime reading input
+syscall			#read it
+blez $v0, retry
+move $s7, $v0 			#size
+mul $s6, $s7, 4 		#4 bytes per word, RAM is byte-accessible
+li $v0, 9 			# 9 allocates $a0 bytes of RAM
+move $a0, $s6 			#we will need to allocate this many bytes
+syscall    			#allocated
```

Comment:
Nice Job! I like the dynamic memory allocation, though an upper bound input check would still be useful because it breaks for n > 46.  If you want the program to be even more efficient, you need no more than 2 registers to store old fibonacci numbers.  Good work in any case.